src/cli.ts

```diff
@@ -59,10 +62,93 @@ const noop = (): void => {
+            let count = 0;
+            await topic.questions?.asyncForEach(async (question)=>{
+                const {webworkQuestionPath, errors, courseQuestionAssessmentInfo} = question;
+                const additionalProblemPaths = courseQuestionAssessmentInfo?.additionalProblemPaths;
+                const additionalErrors = courseQuestionAssessmentInfo?.errors;
+
+                const isDefinitelyBad = VALID_PG_PATH_REGEX.test(webworkQuestionPath) === false;
+
+                // If this is definitely bad and not already in the error object, add it.
+                if (isDefinitelyBad && (_.isNil(errors) || _.isEmpty(errors) || !(webworkQuestionPath in errors))){
+                    logger.debug(`Updating ${question.id} from topic ${topic.id} with a path error.`);
+                    question.errors = {
+                        [webworkQuestionPath]: [`${webworkQuestionPath} cannot be found.`]
+                    };
+                    question.save();
+                }
+
+                if (topic.topicTypeId === 2) {
+                    if (_.isNil(courseQuestionAssessmentInfo)) {
+                        logger.error(`Topic ${topic.id} is an exam but this question has no assessment info.`);
+                        if (isDefinitelyBad) count += 1;
+                        return;
+                    }
+                    let atLeastOneIsBad = false;
+                    await additionalProblemPaths?.asyncForEach(async (path) => {
+                        const isDefinitelyBad = VALID_PG_PATH_REGEX.test(path) === false;
+                        if (isDefinitelyBad) atLeastOneIsBad = true;
+                        
+                        if (isDefinitelyBad && (_.isNil(additionalErrors) || _.isEmpty(additionalErrors) || !(path in additionalErrors))) {
+                            if (_.isNil(courseQuestionAssessmentInfo.errors)) {
+                                courseQuestionAssessmentInfo.errors = {};
+                            }
+                            logger.debug(`Updating ${courseQuestionAssessmentInfo.id} from topic ${topic.id} with an additional path error.`);
+                            courseQuestionAssessmentInfo.errors[path] = [`${path} cannot be found`];
+                        }
+                    });
+                    courseQuestionAssessmentInfo.save();
+                    if (isDefinitelyBad || atLeastOneIsBad) count += 1;
+                } else {
+                    if (isDefinitelyBad) count += 1;
+                }
+            });
+
+            // Count can be equal to or less than because it doesn't include full renderer checks.
+            // Questions will still be annotated correctly with updated error paths.
+            if (count > topic.errors) {
+                logger.debug(`Updating topic ${topic.id} error count from ${topic.errors} to ${count}`);
+                topic.errors = count;
+                topic.save();
```

Comment:
Not awaiting

Reply:
If we don't await we should have a .catch to avoid orphaned promise errors

Reply:
Does that still work for cancelling the transaction?

Reply:
I don't think it will cancel the transaction period since it is orphaned, it might 1 level deep i'm not sure, why don't we await this?